lib/id3c/utils.py

```diff
@@ -14,3 +16,46 @@ def wrap(function):
         function.__doc__ = function.__doc__.format_map(kwargs)
         return function
     return wrap
+
+
+def getattrpath(value: Any, attrpath: Union[str, Sequence[str]]) -> Any:
+    """
+    Get a nested named attribute, described by *attrpath*, from the
+    object *value*, or return None if one of the attributes in the path
+    doesn't exist.
+
+    *attrpath* may be a dotted-string like ``a.b.c`` or a sequence
+    (list or tuple) like ``("a", "b", "c")``.
+
+    >>> class Namespace:
+    ...     def __init__(self):
+    ...         self.__dict__ = {}
+    ...     def __repr__(self):
+    ...         return repr(self.__dict__)
+
+    >>> obj = Namespace()
+    >>> obj.a = Namespace()
+    >>> obj.a.b = Namespace()
+    >>> obj.a.b.c = 42
+    >>> getattrpath(obj, "a")
+    {'b': {'c': 42}}
+    >>> getattrpath(obj, "a.b.c")
+    42
+    >>> getattrpath(obj, "a.b.x")
+    >>> getattrpath(obj, "a.x")
+    >>> getattrpath(obj, "x")
+    >>> getattrpath(obj, ("a", "b", "c"))
+    42
+    """
+    if isinstance(attrpath, str):
+        attrpath = attrpath.split(".")
+
+    assert len(attrpath) != 0
```

Comment:
Could you also say `assert attrpath` here?

This is less of a suggestion and more of my wondering if I'm not considering any edge cases.

Reply:
`assert attrpath` would do the same thing, but explicitly checking length felt more descriptive to me in this case.